bin/check-log.rb

```diff
@@ -128,6 +128,13 @@ class CheckLog < Sensu::Plugin::Check::CLI
          proc: proc(&:to_i),
          default: 250
 
+  option :encode_utf16,
```

Comment:
@majormoses  

> "A regexp can be matched against a string when they either share an encoding, or the regexp’s encoding is US-ASCII and the string’s encoding is ASCII-compatible." 
[[ruby-doc]](https://ruby-doc.org/core-2.2.3/Regexp.html)

The issue that we have is invalid byte sequence when matching strings, and the prefered encoding in such cases is UTF-8 (ASCII is a compatible subset).  
Since we want to remove invalid byte sequence, we are using the encode utf16 to do that for us. 

> "it is impossible to fix an invalid UTF-8 filename using a UTF-16 .... The opposite is not true: it is trivial to translate invalid UTF-16 to a unique (though technically invalid) UTF-8 string, so a UTF-8 API can control both UTF-8 and UTF-16 files and names"
[[wiki]](http://bit.ly/2uHHJvd)

 
Hence we do want to re-encode with utf-8 before calling the _match_ function. 
And we do keeping the default of UTF-8 to avoid breaking changes. 👍